h/static/scripts/directive/excerpt.coffee

```diff
@@ -0,0 +1,48 @@
+###*
+# @ngdoc directive
+# @name excerpt
+# @restrict AE
+# @description Checks to see if text is overflowing its container.
+# If so, it prepends/appends expanders/collapsers. Note, to work
+# the element needs a max height.
+###
+module.exports = ->
+  link: (scope, elem, attr, ctrl) ->
+    scope.collapsed = true
+
+    scope.isOverflowing = ->
+      excerpt = elem[0].querySelector('.excerpt')
+      if excerpt == null
+        return false
+      return elem[0].scrollHeight > elem[0].clientHeight
+
+    scope.toggle = ->
+      scope.collapsed = !scope.collapsed
+
+  scope:
+    enabled: '=excerptIf'
+    maxheight: '=maxheight'
```

Comment:
Funny enough, that's what I was going to do, but worried it wouldn't be modular enough.